concent_api/core/views.py

```diff
@@ -1241,6 +1246,684 @@ def store_message_and_message_status(
+    ack_report_computed_task:       message.AckReportComputedTask        = None,
+    reject_report_computed_task:    message.RejectReportComputedTask     = None,
+    subtask_results_accepted:       message.tasks.SubtaskResultsAccepted = None,
+    subtask_results_rejected:       message.tasks.SubtaskResultsRejected = None,
+):
+    """
+    Validates and stores subtask and its data in Subtask table.
+    Stores related messages in StoredMessage table and adds relation to newly created subtask.
+    """
+    assert isinstance(task_id,              str)
+    assert isinstance(subtask_id,           str)
+    assert isinstance(provider_public_key,  bytes)
+    assert isinstance(requestor_public_key, bytes)
+    assert state in Subtask.SubtaskState
+    assert (state in Subtask.ACTIVE_STATES)  == (next_deadline is not None)
+    assert (state in Subtask.PASSIVE_STATES) == (next_deadline is None)
+
+    provider  = Client.objects.get_or_create_full_clean(provider_public_key)
+    requestor = Client.objects.get_or_create_full_clean(requestor_public_key)
+
+    subtask = Subtask(
+        task_id         = task_id,
+        subtask_id      = subtask_id,
+        provider        = provider,
+        requestor       = requestor,
+        state           = state.name,
+        next_deadline   = parse_timestamp_to_utc_datetime(next_deadline),
+    )
+
+    set_subtask_messages(
+        subtask,
+        task_to_compute             = task_to_compute,
+        report_computed_task        = report_computed_task,
+        ack_report_computed_task    = ack_report_computed_task,
+        reject_report_computed_task = reject_report_computed_task,
+        subtask_results_accepted    = subtask_results_accepted,
+        subtask_results_rejected    = subtask_results_rejected,
+    )
+
+    subtask.full_clean()
+    subtask.save()
+
+    logging.log_subtask_stored(
+        task_id,
+        subtask_id,
+        state.name,
+        subtask.provider.public_key,
+        subtask.requestor.public_key,
+        next_deadline,
+    )
+
+    return subtask
+
+
+def verify_file_status(
+    client_public_key: bytes,
+):
+    """
+    Function to verify existence of a file on cluster storage
+    """
+
+    force_get_task_result_list = PendingResponse.objects.filter(
+        Q(subtask__requestor__public_key = b64encode(client_public_key)) | Q(subtask__provider__public_key = b64encode(client_public_key)),
+        subtask__state                   = Subtask.SubtaskState.FORCING_RESULT_TRANSFER.name,  # pylint: disable=no-member
+    )
+
+    for pending_reponse in force_get_task_result_list:
+        report_computed_task    = deserialize_message(pending_reponse.subtask.report_computed_task.data.tobytes())
+        file_transfer_token     = create_file_transfer_token(
+            report_computed_task,
+            client_public_key,
+        )
+        if get_file_status(file_transfer_token):
```

Comment:
I would suggest either:
1) renaming method as we do not get aything, rather check some state, i.e. `is_file_status_ok`
2) get the actual status and compare if with
As the sttatus is just boolean flag, I would reccomend 1st solution.

Reply:
Yeah, this bothered me a bit too but I had so many other comments that I decided to let this one slide :)

I'd also say that if a function is doing a "heavy" operation (and I consider an external request that can block for a significant time quite heavy in this context) the name should reflect that. Both `get_...` and `is_...` to me suggest a simple getter or a boolean check that do not do much beyond getting a value of some variable. That's why, if I were writing it myself, I'd go with the second approach and call it something other than `get_...`. For example `request_upload_status()`.

Reply:
done